pkg/ssh/agent/agent_windows.go

```diff
@@ -0,0 +1,80 @@
+package agent
+
+import (
+	"io"
+	"os"
+	"sync"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+	gosshagent "golang.org/x/crypto/ssh/agent"
+	"gopkg.in/natefinch/npipe.v2"
+)
+
+const (
+	channelType      = "auth-agent@openssh.com"
+	defaultNamedPipe = "\\\\.\\pipe\\openssh-ssh-agent"
+)
+
+func GetSSHAuthSocket() string {
+	sshAuthSocket := os.Getenv("SSH_AUTH_SOCK")
+	if sshAuthSocket != "" {
+		return sshAuthSocket
+	}
+	if _, err := os.Stat(defaultNamedPipe); err == nil {
+		return defaultNamedPipe
+	}
+
+	return ""
+}
```

Comment:
I'm planning to test this just a little bit more tomorrow, but presently I know this works with the OpenSSH agent that ships with Windows 10/11 natively. I'm pretty sure that the Git for Windows variation utilizes a unix socket which would mean a slight adjustment to the agent forwarding to ensure the correct socket type is used for the connection.

Reply:
I've confirmed that Git for Windows is in fact using an `AF_UNIX` socket, so I'll update the forward method just a little bit to handle that.

Reply:
After some more digging after being puzzled why I couldn't connect using the AF_UNIX socket type as directly utilized by the upstream `golang.org/x/crypto/SSH/agent`, I've come to find out that MSYS2 which Got for Windows is based on uses an emulated socket rather than a native AF_UNIX socket as I was led to believe. 

Now with this said, there are a myriad of authenticators that do create valid AF_UNIX sockets for Windows and I found a PR for Win32-OpenSSH to add native AF_UNIX sockets to the build that comes packaged with Windows. Unfortunately, [that PR](https://github.com/PowerShell/openssh-portable/pull/674) is not yet merged.

What I think I'm going to do here is to prefer a set SSH_AUTH_SOCK, this way various correct implementations work, then fallback to the named pipe implementation, then finally return an error. When Windows finally natively supports SSH_AUTH_SOCK definitions, then the code will natively migrate to that updated path this way.

I'll submit this small change tomorrow, but then this should be good to go.